text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,489 @@
++ Feature name: `iri_v181_transaction_and_bundle`
++ Start date: 2019-09-06
++ RFC PR: [iotaledger/bee-rfcs#3](https://github.com/iotaledger/bee-rfcs/pull/3),
+[iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18),
+[iotaledger/bee-rfcs#20](https://github.com/iotaledger/bee-rfcs/pull/20)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+The fundamental communication unit in the IOTA protocol is the transaction. Everything, including payment settlements
+and plain data, is propagated through the IOTA network in transactions.
+
+A transaction is `8019` trits and the payload - `sig_or_msg` field - is `6561` trits. This payload can hold a signature
+fragment or a message fragment. Since it has a limited size, a user often needs more than one transaction to fulfil
+their operation, for example signatures with security level `2` or `3` don't fit in a single transaction and
+user-provided message may exceed the allowance so they need to be fragmented across multiple transactions. Moreover,
+input/output transactions doesn't make sense on their own because they would change the total amount of the ledger so
+they have to be paired with other input/output transactions that together will balance the total value to zero.
+
+For these reasons, transactions have to be processed as a whole, in groups called bundles. A bundle is an atomic
+operation in the sense that either all or none of its transactions are accepted by the network. Even single
+transactions are propagated through the network within a bundle.
+
+By analogy with TCP, a bundle corresponds to a stream, and a transaction corresponds to a packet.
```

Comment:
Are we expecting RFC readers to be knowledgeable in TCP to understand this analogy? Apart from that the analogy is bad because a bundle is not a stream (which insinuates potential endlessness), it's much more similar to a container.